src/PrestaShopBundle/Resources/config/services/adapter/product/_image.yml

```
@@ -71,7 +72,19 @@ services:
         class: PrestaShop\PrestaShop\Adapter\Product\Image\ProductImagePathFactory
         arguments:
             - "@=service('prestashop.adapter.legacy.configuration').getBoolean('PS_LEGACY_IMAGES')"
+            - ''
```

```suggestion
            - !php/const _PS_PROD_IMG_DIR_
```

This one should be used by the ImageUploader and needs the absolute **file system path** or else this will beak its initial behaviour along with Image upload.

I'm now sure we should rename this service id to avoid any confusion so I'd suggest `prestashop.adapter.product.image.product_image_filesystem_path_factory`

---

My bad!!
it should be `%kernerl.project_dir%` instead